src/filter/json_encode_filter.rs

```diff
@@ -0,0 +1,133 @@
+//! Convert LogLine events into Raw events encoded as JSON.
+//!
+//! This filter takes LogLines and encodes them into JSON, emitting the encoded
+//! event as a Raw event. This allows further filters or sinks to operate on the
+//! JSON without needing to understand a LogLine event in particular.
+//! If the LogLine value is a valid JSON object and parse_line config option is true,
+//! then the JSON will be merged with LogLine metadata. Otherwise, the original line
+//! will be included simply as a string.
+
+use chrono::DateTime;
+use chrono::naive::NaiveDateTime;
+use chrono::offset::Utc;
+use filter;
+use metric;
+use metric::TagMap;
+use rand::random;
+use serde_json;
+use serde_json::Value;
+use serde_json::map::Map;
+use std::iter::FromIterator;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+lazy_static! {
+    /// Total number of logline processed
+    pub static ref JSON_ENCODE_LOG_PROCESSED: Arc<AtomicUsize> = Arc::new(AtomicUsize::new(0));
+    /// Total number of logline with JSON value successfully parsed
+    pub static ref JSON_ENCODE_LOG_PARSED: Arc<AtomicUsize> = Arc::new(AtomicUsize::new(0));
+}
+
+impl From<TagMap> for Map<String, Value> {
+    fn from(tagmap: TagMap) -> Self {
+        Map::from_iter(
+            tagmap.into_iter().map(|(k, v)| (k.clone(), v.clone().into()))
+        )
+    }
+}
+
+/// Convert LogLine events into Raw events encoded as JSON.
+///
+/// This filter takes LogLines and encodes them into JSON, emitting the encoded
+/// event as a Raw event. This allows further filters or sinks to operate on the
+/// JSON without needing to understand a LogLine event in particular.
+/// If the LogLine value is a valid JSON object and parse_line config option is true,
+/// then the JSON will be merged with LogLine metadata. Otherwise, the original line
+/// will be included simply as a string.
+pub struct JSONEncodeFilter {
+    parse_line: bool,
+}
+
+/// Configuration for `JSONEncodeFilter`
+#[derive(Clone, Debug)]
+pub struct JSONEncodeFilterConfig {
+    /// The filter's unique name in the routing topology.
+    pub config_path: Option<String>,
+    /// The forwards along which the filter will emit its `metric::Event`s.
+    pub forwards: Vec<String>,
+    /// Whether the filter should attempt to parse LogLine values that are valid JSON objects.
+    pub parse_line: bool,
+}
+
+impl JSONEncodeFilter {
+    /// Create a new JSONEncodeFilter
+    pub fn new(config: &JSONEncodeFilterConfig) -> JSONEncodeFilter {
+        JSONEncodeFilter {
+            parse_line: config.parse_line,
+        }
+    }
+}
+
+impl filter::Filter for JSONEncodeFilter {
+    fn process(
+        &mut self,
+        event: metric::Event,
+        res: &mut Vec<metric::Event>,
+    ) -> Result<(), filter::FilterError> {
+        match event {
+            metric::Event::Telemetry(_) => {
+                // Drop telemetry
```

Comment:
Rather than drop we should probably forward Telemetry through.

Reply:
fair enough. wasn't sure of the convention for filters that don't apply to a particular type.

Reply:
We don't have a concrete policy around this and we should. I lean toward "pass forward unless otherwise documented".